tests/Microsoft.Identity.Test.Unit/BrokerTests/RuntimeBrokerTests.cs

```diff
@@ -153,7 +187,116 @@ public void HandleInstallUrl_Throws()
+        // Done - Is Pii enabled
+        // Done - does not emit when loglevel is not appropriate
+        // Is API Platform logging enabled?
+        // no mem leaks
+
+        [TestMethod]
+        public async Task ATS_CallsLog_When_CalledAsync()
+        {
+            // Arrange
+            var appTokenCache = new TokenCache(_serviceBundle, isApplicationTokenCache: true);
+            var requestContext = new RequestContext(_serviceBundle, Guid.NewGuid());
+            var tenantAuthority = AuthorityInfo.FromAadAuthority(AzureCloudInstance.AzurePublic, tenant: TestConstants.AadTenantId, validateAuthority: false);
+            var acquireTokenCommonParameters = new AcquireTokenCommonParameters
+            {
+                ApiId = ApiEvent.ApiIds.AcquireTokenForClient,
+                AuthorityOverride = tenantAuthority
+            };
+
+            var authRequestParams = new AuthenticationRequestParameters(
+                _serviceBundle,
+                appTokenCache,
+                acquireTokenCommonParameters,
+                requestContext,
+                Authority.CreateAuthority(tenantAuthority));
+            authRequestParams.Scope.Add("User.Read");
+
+            AcquireTokenSilentParameters silentParams = new AcquireTokenSilentParameters();
+            silentParams.Account = PublicClientApplication.OperatingSystemAccount;
+
+            // Act
+            try
+            {
+                _ = await _wamBroker.AcquireTokenSilentAsync(authRequestParams, silentParams).ConfigureAwait(false);
+            }
+            catch (MsalUiRequiredException)
+            {
+                // Assert
+                _logger.Received().Log(Arg.Any<Client.LogLevel>(), Arg.Any<string>(), Arg.Any<string>());
+            }
+            catch (Exception ex)
+            {
+                Assert.IsTrue(false, ex.Message);
+            }
+        }
+
+        [DataTestMethod]
+        [DataRow(Client.NativeInterop.LogLevel.Trace, Client.LogLevel.Verbose)]
+        [DataRow(Client.NativeInterop.LogLevel.Debug, Client.LogLevel.Verbose)]
+        [DataRow(Client.NativeInterop.LogLevel.Info, Client.LogLevel.Info)]
+        [DataRow(Client.NativeInterop.LogLevel.Warning, Client.LogLevel.Warning)]
+        [DataRow(Client.NativeInterop.LogLevel.Error, Client.LogLevel.Error)]
+        [DataRow(Client.NativeInterop.LogLevel.Fatal, Client.LogLevel.Error)]
+        public void LogEventRaised_MapsEvents_Correctly_When_No_Pii(Client.NativeInterop.LogLevel nativeLogLevel, Client.LogLevel msalLogLevel)
+        {
+            const string logMessage = "This is test";
+            _logger.IsLoggingEnabled(msalLogLevel).Returns(true);
+
+            Type wamBrokerType = _wamBroker.GetType();
+            MethodInfo fireLogMethod = wamBrokerType.GetMethod("LogEventRaised", BindingFlags.NonPublic | BindingFlags.Instance);
+            fireLogMethod.Invoke(_wamBroker, new object[] { null, new LogEventArgs(nativeLogLevel, logMessage) });
```

Comment:
Any way to test this without reflection? Make LogEventRaised internal?

Reply:
I do not want to change the access modifier just of testing. If we can do testing with reflection, I avoid it.